zerver/lib/integrations.py

```diff
@@ -186,6 +186,40 @@ def url_object(self):
         # type: () -> None
         return
 
+class EmbeddedBotService(Integration):
+    '''
+    This class acts as a registry for bots verified as safe
+    and valid such that these are capable of being deployed on the server.
+    '''
+    IS_ALLOWED = True
+    # This variable informs if this embedded bot service is allowed to run by this
+    # community members. The Organization administrators can turn this off if they
+    # do not want community members to run this bot service.
+
+    def __init__(self, name):
+        # type: (str) -> None
+        super(EmbeddedBotService, self).__init__(
+            name=name,
+            client_name=name,
+            # TODO: Not sure how client_name would be useful here, but it is not None parameter.
+            # setting it to name of the bot for now.
```

Comment:
I think we want to generate a `client_name` based on the name of the bot, similar to what we do with webhook integrations.  E.g. this might want to be `ZulipWikipedaBot`.